diff -r -u old/pango-1.36.3/pango/pangocoretext.c new/pango-1.36.3/pango/pangocoretext.c
--- old/pango-1.36.3/pango/pangocoretext.c	2014-03-05 21:33:55.000000000 -0700
+++ new/pango-1.36.3/pango/pangocoretext.c	2014-04-14 18:25:38.000000000 -0600
@@ -97,8 +97,7 @@
   bitmap = CFCharacterSetCreateBitmapRepresentation (kCFAllocatorDefault,
                                                      charset);
 
-  /* We only handle the BMP plane */
-  length = MIN (CFDataGetLength (bitmap), 8192);
+  length = CFDataGetLength (bitmap);
   ptr = CFDataGetBytePtr (bitmap);
 
   /* FIXME: can and should this be done more efficiently? */
diff -r -u old/pango-1.36.3/modules/basic/basic-coretext.c new/pango-1.36.3/modules/basic/basic-coretext.c
--- old/pango-1.36.3/modules/basic/basic-coretext.c	2014-03-05 21:33:55.000000000 -0700
+++ new/pango-1.36.3/modules/basic/basic-coretext.c	2014-04-14 17:50:38.000000000 -0600
@@ -55,7 +55,8 @@
 	   PangoGlyphString *glyphs,
 	   int               i,
 	   int               offset,
-	   PangoGlyph        glyph)
+	   PangoGlyph        glyph,
+           CGSize            *adv)
 {
   PangoRectangle logical_rect;
 
@@ -65,8 +66,13 @@
   glyphs->glyphs[i].geometry.y_offset = 0;
 
   glyphs->log_clusters[i] = offset;
-  pango_font_get_glyph_extents (font, glyphs->glyphs[i].glyph, NULL, &logical_rect);
-  glyphs->glyphs[i].geometry.width = logical_rect.width;
+  if (adv) {
+    /* by using the advances array, we get kerning */
+    glyphs->glyphs[i].geometry.width = adv->width * PANGO_SCALE;
+  } else {
+    pango_font_get_glyph_extents (font, glyphs->glyphs[i].glyph, NULL, &logical_rect);
+    glyphs->glyphs[i].geometry.width = logical_rect.width;
+  }
 }
 
 
@@ -92,6 +98,7 @@
   CTRunRef current_run;
   CFIndex *current_indices;
   const CGGlyph *current_cgglyphs;
+  const CGSize *current_cgadvs;
   CTRunStatus current_run_status;
 };
 
@@ -118,6 +125,7 @@
   iter->current_run = CFArrayGetValueAtIndex (iter->runs, run_number);
   iter->current_run_status = CTRunGetStatus (iter->current_run);
   iter->current_cgglyphs = CTRunGetGlyphsPtr (iter->current_run);
+  iter->current_cgadvs = CTRunGetAdvancesPtr (iter->current_run);
 
   ct_glyph_count = CTRunGetGlyphCount (iter->current_run);
   iter->current_indices = malloc (sizeof (CFIndex *) * ct_glyph_count);
@@ -166,7 +174,17 @@
 static gunichar
 run_iterator_get_character (struct RunIterator *iter)
 {
-  return CFStringGetCharacterAtIndex (iter->cstr, iter->current_indices[iter->ct_i]);
+  gunichar c;
+
+  c = CFStringGetCharacterAtIndex (iter->cstr, iter->current_indices[iter->ct_i]);
+  if ((c >= 0xD800) && (c <= 0xDFFF)) {
+    /* surrogate pair */
+    gunichar c2;
+    c2 = CFStringGetCharacterAtIndex (iter->cstr, iter->current_indices[iter->ct_i]+1);
+    c = 0x10000 + (((c & 0x3FF) << 10) | (c2 & 0x3FF));
+  }
+
+  return c;
 }
 
 static CGGlyph
@@ -175,6 +193,12 @@
   return iter->current_cgglyphs[iter->ct_i];
 }
 
+static CGSize
+run_iterator_get_cgadv (struct RunIterator *iter)
+{
+  return iter->current_cgadvs[iter->ct_i];
+}
+
 static CFIndex
 run_iterator_get_index (struct RunIterator *iter)
 {
@@ -297,6 +321,7 @@
 {
   CFIndex index;
   CGGlyph cgglyph;
+  CGSize cgadv;
   gunichar wc;
 };
 
@@ -338,6 +363,7 @@
       gi = g_slice_new (struct GlyphInfo);
       gi->index = run_iterator_get_index (&riter);
       gi->cgglyph = run_iterator_get_cgglyph (&riter);
+      gi->cgadv = run_iterator_get_cgadv (&riter);
       gi->wc = run_iterator_get_character (&riter);
 
       glyph_list = g_slist_prepend (glyph_list, gi);
@@ -426,7 +452,7 @@
       if (gi == NULL || gi->index > gs_i)
         {
           /* gs_i is behind, insert empty glyph */
-          set_glyph (font, glyphs, gs_i, p - text, PANGO_GLYPH_EMPTY);
+          set_glyph (font, glyphs, gs_i, p - text, PANGO_GLYPH_EMPTY, NULL);
           continue;
         }
       else if (gi->index < gs_i)
@@ -457,7 +483,7 @@
 
           if (result != PANGO_COVERAGE_NONE)
             {
-              set_glyph (font, glyphs, gs_i, p - text, gi->cgglyph);
+              set_glyph (font, glyphs, gs_i, p - text, gi->cgglyph, &gi->cgadv);
 
               if (g_unichar_type (gi->wc) == G_UNICODE_NON_SPACING_MARK)
                 {
@@ -480,7 +506,7 @@
                 }
             }
           else
-            set_glyph (font, glyphs, gs_i, p - text, PANGO_GET_UNKNOWN_GLYPH (gi->wc));
+            set_glyph (font, glyphs, gs_i, p - text, PANGO_GET_UNKNOWN_GLYPH (gi->wc), NULL);
 
           glyph_iter = g_slist_next (glyph_iter);
         }
diff -r -u old/pango-1.36.3/pango/pangocairo-coretextfont.c new/pango-1.36.3/pango/pangocairo-coretextfont.c
--- old/pango-1.36.3/pango/pangocairo-coretextfont.c	2014-03-05 21:33:55.000000000 -0700
+++ new/pango-1.36.3/pango/pangocairo-coretextfont.c	2014-03-30 09:52:46.000000000 -0600
@@ -147,6 +147,12 @@
   metrics->strikethrough_position = metrics->ascent / 3;
   metrics->strikethrough_thickness = CTFontGetUnderlineThickness (ctfont) * PANGO_SCALE;
 
+  metrics->underline_position = -metrics->underline_position;
+  pango_quantize_line_geometry (&metrics->underline_thickness,
+                                &metrics->underline_position);
+  metrics->underline_position = -(metrics->underline_position 
+                                  + metrics->underline_thickness);
+
   layout = pango_layout_new (context);
   font_desc = pango_font_describe_with_absolute_size ((PangoFont *) font);
   pango_layout_set_font_description (layout, font_desc);
diff -r -u old/pango-1.36.3/pango/pangocairo-coretextfontmap.c new/pango-1.36.3/pango/pangocairo-coretextfontmap.c
--- old/pango-1.36.3/pango/pangocairo-coretextfontmap.c	2014-03-05 21:33:55.000000000 -0700
+++ new/pango-1.36.3/pango/pangocairo-coretextfontmap.c	2014-03-30 09:52:46.000000000 -0600
@@ -186,5 +186,5 @@
 pango_cairo_core_text_font_map_init (PangoCairoCoreTextFontMap *cafontmap)
 {
   cafontmap->serial = 1;
-  cafontmap->dpi = 96.;
+  cafontmap->dpi = 72.;
 }
diff -r -u old/pango-1.36.3/pango/pangocoretext-fontmap.c new/pango-1.36.3/pango/pangocoretext-fontmap.c
--- old/pango-1.36.3/pango/pangocoretext-fontmap.c	2014-04-14 10:56:38.000000000 -0600
+++ new/pango-1.36.3/pango/pangocoretext-fontmap.c	2014-04-14 11:01:45.000000000 -0600
@@ -297,7 +297,8 @@
   cf_number = (CFNumberRef)CFDictionaryGetValue (dict,
                                                  kCTFontWeightTrait);
 
-  if (CFNumberGetValue (cf_number, kCFNumberCGFloatType, &value))
+  weight = PANGO_WEIGHT_NORMAL;
+  if (cf_number && CFNumberGetValue (cf_number, kCFNumberCGFloatType, &value))
     {
       if (value < ct_weight_min || value > ct_weight_max)
 	{
@@ -315,8 +316,6 @@
 	      }
 	}
     }
- else
-   weight = PANGO_WEIGHT_NORMAL;
 
   CFRelease (dict);
 
