diff -r -u old/glib-2.56.0/gio/gdbusauth.c new/glib-2.56.0/gio/gdbusauth.c
--- old/glib-2.56.0/gio/gdbusauth.c	2020-08-18 13:43:21.000000000 -0600
+++ new/glib-2.56.0/gio/gdbusauth.c	2020-08-18 13:44:09.000000000 -0600
@@ -1299,9 +1299,9 @@
                                                     &line_length,
                                                     cancellable,
                                                     error);
-          debug_print ("SERVER: WaitingForBegin, read '%s'", line);
           if (line == NULL)
             goto out;
+          debug_print ("SERVER: WaitingForBegin, read '%s'", line);
           if (g_strcmp0 (line, "BEGIN") == 0)
             {
               /* YAY, done! */
Only in new/glib-2.56.0/gio: gdbusauth.c~
diff -r -u old/glib-2.56.0/gio/gdbusmessage.c new/glib-2.56.0/gio/gdbusmessage.c
--- old/glib-2.56.0/gio/gdbusmessage.c	2020-08-18 13:43:21.000000000 -0600
+++ new/glib-2.56.0/gio/gdbusmessage.c	2020-08-18 13:50:21.000000000 -0600
@@ -2699,6 +2699,7 @@
   if (message->body != NULL)
     {
       gchar *tupled_signature_str;
+      if (signature_str == NULL) signature_str = "???";
       tupled_signature_str = g_strdup_printf ("(%s)", signature_str);
       if (signature == NULL)
         {
Only in new/glib-2.56.0/gio: gdbusmessage.c~
