diff -r -u old/glib-2.56.0/glib/gbacktrace.h new/glib-2.56.0/glib/gbacktrace.h
--- old/glib-2.56.0/glib/gbacktrace.h	2018-01-08 14:34:19.000000000 -0700
+++ new/glib-2.56.0/glib/gbacktrace.h	2022-04-17 19:21:22.000000000 -0600
@@ -58,7 +58,7 @@
 #  define G_BREAKPOINT()        G_STMT_START{ __debugbreak(); }G_STMT_END
 #elif defined (__alpha__) && !defined(__osf__) && defined (__GNUC__) && __GNUC__ >= 2
 #  define G_BREAKPOINT()        G_STMT_START{ __asm__ __volatile__ ("bpt"); }G_STMT_END
-#elif defined (__APPLE__)
+#elif defined (__APPLE__) || (defined(_WIN32) && (defined(__clang__) || defined(__GNUC__)))
 #  define G_BREAKPOINT()        G_STMT_START{ __builtin_trap(); }G_STMT_END
 #else   /* !__i386__ && !__alpha__ */
 #  define G_BREAKPOINT()        G_STMT_START{ raise (SIGTRAP); }G_STMT_END
Only in new/glib-2.56.0/glib: gbacktrace.h~
