diff -r -u old/poppler-0.24.5/libtool new/poppler-0.24.5/libtool
--- old/poppler-0.24.5/libtool	2020-08-18 14:20:55.000000000 -0600
+++ new/poppler-0.24.5/libtool	2020-08-18 14:22:24.000000000 -0600
@@ -10167,14 +10167,14 @@
 old_archive_from_expsyms_cmds=""
 
 # Commands used to build a shared archive.
-archive_cmds="\$CC -shared -nostdlib \$predep_objects \$libobjs \$deplibs \$postdep_objects \$compiler_flags -o \$output_objdir/\$soname \${wl}--enable-auto-image-base -Xlinker --out-implib -Xlinker \$lib"
+archive_cmds="\$CC -shared \$predep_objects \$libobjs \$deplibs \$postdep_objects \$compiler_flags -o \$output_objdir/\$soname \${wl}--enable-auto-image-base -Xlinker --out-implib -Xlinker \$lib"
 archive_expsym_cmds="if test \\\"x\\\`\$SED 1q \$export_symbols\\\`\\\" = xEXPORTS; then
 	      cp \$export_symbols \$output_objdir/\$soname.def;
 	    else
 	      echo EXPORTS > \$output_objdir/\$soname.def;
 	      cat \$export_symbols >> \$output_objdir/\$soname.def;
 	    fi~
-	    \$CC -shared -nostdlib \$output_objdir/\$soname.def \$predep_objects \$libobjs \$deplibs \$postdep_objects \$compiler_flags -o \$output_objdir/\$soname \${wl}--enable-auto-image-base -Xlinker --out-implib -Xlinker \$lib"
+	    \$CC -shared \$output_objdir/\$soname.def \$predep_objects \$libobjs \$deplibs \$postdep_objects \$compiler_flags -o \$output_objdir/\$soname \${wl}--enable-auto-image-base -Xlinker --out-implib -Xlinker \$lib"
 
 # Commands used to build a loadable module if different from building
 # a shared archive.
@@ -10266,2 +10266,7 @@
 
+
+# Clear these back out:
+predep_objects=""
+postdep_objects=""
+postdeps=""
 # ### END LIBTOOL TAG CONFIG: CXX
