diff -u -r old/pixman-0.32.4/pixman/pixman-compiler.h new/pixman-0.32.4/pixman/pixman-compiler.h
--- old/pixman-0.32.4/pixman/pixman-compiler.h	2014-04-04 15:06:39.000000000 -0600
+++ new/pixman-0.32.4/pixman/pixman-compiler.h	2014-04-04 15:07:27.000000000 -0600
@@ -78,7 +78,7 @@
 #   define noinline __declspec(noinline)
 #elif defined __GNUC__ || (defined(__SUNPRO_C) && (__SUNPRO_C >= 0x590))
 #   define inline __inline__
-#   define force_inline __inline__ __attribute__ ((__always_inline__))
+#   define force_inline __inline__
 #   define noinline __attribute__((noinline))
 #else
 #   ifndef force_inline
Only in new/pixman-0.32.4/pixman: pixman-compiler.h~
